From cf504278288b39350ab78fbb058051a01f0e4338 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 10 Aug 2021 12:04:56 +0300
Subject: [PATCH] [SW-0] gaudi: security: fine tune stlb

Open additional register in the STLB space -
mmSTLB_MEM_CACHE_INVALIDATION.
It should be used for invalidating the cache in high level
applications.

Change-Id: Ib129d812658eede7537fab70b5b5673ed366fb81
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 3f27c4d..1abe9c6 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -941,7 +941,8 @@
 	prot_base = base;
 	mask = ~(PROT_BIT(mmSTLB_CACHE_INV) &
 		 PROT_BIT(mmSTLB_INV_PS) &
-		 PROT_BIT(mmSTLB_INV_SET));
+		 PROT_BIT(mmSTLB_INV_SET) &
+		 PROT_BIT(mmSTLB_MEM_CACHE_INVALIDATION));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0xF80 */
